cmd/cored/main.go

```diff
@@ -152,9 +152,26 @@ func main() {
 		internalDN = &x509Cert.Subject
 	}
 
+	// TODO(kr): make core.UseTLS take just an http client
+	// and use this object in it.
+	httpClient := new(http.Client)
+	httpClient.Transport = &http.Transport{
+		TLSClientConfig: tlsConfig,
+
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		MaxIdleConns:          100,
```

Comment:
just wondering, are there reasons we've chosen these numbers? I feel like we set timeouts and things other places too (although now I can't find anything specific), I wonder if we should be more deliberate about these things